clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -762,15 +765,24 @@ public boolean isDone() {
     private static class InterceptorCallback<K, V> implements Callback {
         private final Callback userCallback;
         private final ProducerInterceptors<K, V> interceptors;
+        private final TopicPartition tp;
 
-        public InterceptorCallback(Callback userCallback, ProducerInterceptors<K, V> interceptors) {
+        public InterceptorCallback(Callback userCallback, ProducerInterceptors<K, V> interceptors,
```

Comment:
Good point. Maybe InterceptorAndUserCallback? I am interested to know if anyone else has any ideas. We can create a separate JIRA for it.